domain/application/service/application.go

```diff
@@ -50,7 +50,23 @@ type ApplicationState interface {
 
 	// AddUnits adds the specified units to the application, returning an error
 	// satisfying [applicationerrors.ApplicationNotFoundError] if the application doesn't exist.
-	AddUnits(ctx context.Context, applicationName string, args ...application.AddUnitArg) error
+	AddUnits(ctx context.Context, applicationName string, args ...application.UpsertUnitArg) error
+
+	// GetApplicationID returns the ID for the named application, returning an error
+	// satisfying [applicationerrors.ApplicationNotFound] if the application is not found.
+	GetApplicationID(ctx context.Context, name string) (coreapplication.ID, error)
```

Comment:
I notice that GetApplicationID takes its own `(ctx context.Context)`
But it is being called from inside `RunAtomic`.